src/odla_layer.c

```diff
@@ -80,13 +88,24 @@ odla_layer make_odla_layer(int batch, int h, int w, int c,
     l.num_input = odla_num_input(l.odla_runtime);
     l.input_tensors = calloc(l.num_input, sizeof(tensor));
     for (i = 0; i < l.num_input; i++) {
-        l.input_tensors[i].w = odla_input_width(l.odla_runtime, i);
-        l.input_tensors[i].h = odla_input_height(l.odla_runtime, i);
-        l.input_tensors[i].c = odla_input_channel(l.odla_runtime, i);
-
-        l.input_tensors[i].size = odla_input_size(l.odla_runtime, i);
-        l.input_tensors[i].buffer = NULL;
-        odla_alloc_input_tensor(l.odla_runtime, &l.input_tensors[i].buffer, i);
+        if (prev_odla_layer == NULL) {
```

Comment:
this will avoid allocating all input tensors. what if we want to allocate+bind one tensor and just bind another tensor?

We need one more param for indicating input tensor in odla layer.
- from_odla_layer
- from_odla_tensor
- to_odla_tensor

Keep everything same, except below check

if (i == to_dla_tensor)
   odla_bind_input_tensor
else
   odla_alloc_input_tensor

Reply:
Fixed the issue.
if (prev_odla_layer != NULL && i == l.input_tensor)
    bind only
else
    alloc + bind